### PR TITLE
Improve collapsing of whitespace on paste in slate block

### LINKED ISSUE #3802: Pasting specially crafted HTML with inner `\n`  in a slate block using FF inserts spaces at the beginning and (apparently) randomly in the middle of the text
**Describe the bug**
If the copied HTML is crafted in a way that it contains `\n` on it, pasting in a slate block using FF inserts soft spaces at the beginning and (apparently) randomly in the middle of the text.

**To Reproduce**

1. Using FF
2. Use https://www.dlr.de/content/de/artikel/news/2022/03/20220831_intelligentes-lademanagement.html
3. Copy the text
4. Paste in a slate block 

See the initial spaces in each block, also some weird double ones in the middle.

image

**Expected behavior**
The pasted content is clear of glitches.

### PR DESCRIPTION
**Pull request by tiberiuichim**
Fixes #3802

### CODE REVIEW

File: packages/volto-slate/src/editor/deserialize.js [hunk 1]
```diff
@@ -56,8 +56,12 @@ export const deserialize = (editor, el) => {
         : null;
     }
     return {
+      // Convert the text to "browser display". Browsers will collapse
+      // whitespaces to a single space, so let's try to reproduce that behavior
       text: el.textContent
-        .replace(/\n$/g, ' ')
+        .replace(/^\n(?=\S)/g, '') // remove "\n" followed by any non-space character
+        .replace(/\n(?=\s)/g, '') // remove \n followed by a space
+        .replace(/(?<=\s)\n/g, '') // remove \n follows a space
```
Comment (davisagli) [inline packages/volto-slate/src/editor/deserialize.js:64]: @tiberiuichim not supported in Safari: https://caniuse.com/js-regexp-lookbehind

Instead you could do `(\s)\n` and replace it with the captured character

File: packages/volto-slate/src/editor/extensions/insertData.js [hunk 2]
```diff
@@ -29,6 +29,9 @@ export const insertData = (editor) => {
 
       let fragment;
 
+      // console.log('dt', dt);
+      // console.log('body', body);
+
```
Comment (davisagli) [inline packages/volto-slate/src/editor/extensions/insertData.js:34]: ?

**Comment by davisagli** [comment]
@tiberiuichim Might be worth having a few tests?

**Review by stevepiercy — CHANGES_REQUESTED** [review]
Fix name and spell out JavaScript

**Comment by tiberiuichim** [comment]
@sneridagh Please test this in client project

**Comment by sneridagh** [comment]
@tiberiuichim confirmed the two bugs (the space leading the pasted text and the other with the double spaces) are gone using this PR

**Review by sneridagh — APPROVED** [review]
@tiberiuichim looks good for me. 

The problem is gone.

Do we want to leave the console.debug statements?

**Comment by tiberiuichim** [comment]
@sneridagh no, the console.debug calls stay. We need them in case we encounter more weird pasting behavior. They shouldn't affect regular users.

**Comment by tiberiuichim** [comment]
@sneridagh Assuming tests pass, this should be good to go.

**Pull request merged**